### DISCUSSION TITLE
[@types/react-burger-menu] version: 2.8.7 Namespace 'React' has no exported member 'JSX'

### DISCUSSION THREAD

**Discussion by tk109o** [DefinitelyTyped/DefinitelyTyped#69731] [Issues with a @types package] [answered]
Hello.

After installing @types/react-burger-menu version 2.8.7, running `npm run server:build`
results in the following error.

```
node_modules/@types/react-burger-menu/index.d.ts:31:30 - error TS2694: Namespace 'React' has no exported member 'JSX'.

31     customBurgerIcon?: React.JSX.Element | false | undefined;
```

It has been confirmed that @types/react functions correctly in the version specified in the PR below.

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64464
> These diffs will work on all latest major types versions e.g. freshly installed @types/react@latest, @types/react@^17.0.0 etc.

It seems the error occurs because JSX.Element is not present in the React type definitions. Is this the expected behavior?

If this is the expected behavior, how should the code be modified?

It appears to be fixed in the following PR, and it has been confirmed that it works with version 2.8.6.
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67641

Please find the package.json and the relevant source code attached below.
(Since this is a translation, there might be some awkward phrasing. Sorry about that.)

package.json
```json
 "scripts": {
  "server:build": "tsc server.ts --esModuleInterop"
  },
 "dependencies": {
     "next": "12.0.4",
  },
 "devDependencies": {
    "@types/react": "^17.0.14",
    "@types/react-dom": "^17.0.9",
   "typescript": "^4.3.5"
 }
```

```tsx
// components/Header.tsx

import React, {MouseEventHandler, useState} from "react"
import { slide as Menu } from "react-burger-menu"
const Header: React.VFC<Props> = ({hoge, foo}) => {
 ...
  return (
        <Menu 
          width={ "20%" }
          customBurgerIcon={ isNavMenuOpen ? undefined : <img src="menuIcon.png" /> }
          customCrossIcon={ <img src="closedIcon.png" /> }
          onStateChange={({ isOpen }) => setIsNavMenuOpen(isOpen)}
          styles={ { bmBurgerButton: commonButtonStyles, bmCrossButton: commonButtonStyles } }
        >
        </Menu>
  )

  }
```

[a comment by an automated account is not shown]

**Comment by dacevedo12** [comment, marked as answer]
Hi, it seems you are using older versions of react and next.js, and as such you have two options: either upgrade to a newer version of next.js/react or use an older version of react-burger-menu.

the JSX.Element global was indeed deprecated in recent versions of react and the right thing is to use React.JSX.Element as done in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67641
